mimic.py

```diff
@@ -46,25 +46,60 @@
 import random
 import sys
 
+def read_file_words( filename ):
+    with open( filename, 'r' ) as f:
+        return f.read().split()
 
-def mimic_dict(filename):
+
+def find_indexes( word, word_list ):
+    index_list = []
+    for ( i, v ) in enumerate( word_list, 1 ):
+        if v.lower() == word.lower():
+            if i < len( word_list ):
+                index_list.append( i )
+            else:
+                pass
+        else:
+            pass
+    return index_list
+
+
+def mimic_dict( filename ):
     """Returns mimic dict mapping each word to list of words which follow it."""
     # +++your code here+++
-    raise NotImplementedError("Get to Work!")
+    # Read the file and return wordlist.
+    words = read_file_words( filename )
+    # get keys for dict
+    keys = list( set( words ) )
+    # Iterate through the keys and create dict
+    dictionary = {}
+    for key in keys:
+        index_list = find_indexes( key, words )
+        dictionary[ key ] = filter( lambda w: words.index( w ) in index_list, words )
```

Comment:
I'm getting a lot of hang time when this runs and I think it has something to do with you find_indexes function in combination with how your dictionary is built.

If you build your dictionary such that each key is a word with a value of a list contained each word that follows that word in the document and then you random.choice from that list I think you'll have better success.

Eg. "the": ['rabbit', 'rabbit', 'hill', 'forest', etc...]

Currently it's crashing on me around line 78-79.

Reply:
Hey I fixed this a while ago.
```suggestion
        dictionary[ key ] = filter( lambda w: words.index( w ) in index_list, words )ey 
```